src/solvers/floatbv/float_utils.cpp

```diff
@@ -152,6 +152,36 @@ bvt float_utilst::build_constant(const ieee_float_valuet &src)
   return pack(bias(result));
 }
 
+bvt float_utilst::round_to_integral(const bvt &src)
+{
+  PRECONDITION(src.size() == spec.width());
+
+  // Zero? NaN? Infinity?
+  auto unpacked = unpack(src);
+  auto is_special = prop.lor({unpacked.zero, unpacked.NaN, unpacked.infinity});
+
+  // add 2^f, where f is the number of fraction bits,
+  // by adding f to the exponent
+  auto magic_number = ieee_floatt{
+    spec, ieee_floatt::rounding_modet::ROUND_TO_ZERO, power(2, spec.f)};
+
+  auto magic_number_bv = build_constant(magic_number);
+
+  // abs(x) >= magic_number? If so, then there is no fractional part.
+  literalt ge_magic_number = relation(abs(src), relt::GE, magic_number_bv);
+
+  magic_number_bv.back() = src.back(); // copy sign bit
+
+  auto tmp1 = add_sub(src, magic_number_bv, false);
```

Comment:
Can this overflow when the number of exponent bits is small?

Reply:
In that case, `ge_magic_number` will be true.